code/script.js

```diff
@@ -222,104 +243,151 @@ const generateBoard = () => {
   })
 }
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
+// Randomly selected character from characters array. Set as the value of the global variable called secret
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+console.log(secret)
+};
+
+// close overlay --> "play again button"
+const closeOverlay = () => {
+  wrapperOverlay.style.display = "none";
+};
 
-// This function to start (and restart) the game
+// board when start & restart game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  charactersInPlay = CHARACTERS 
+  generateBoard();  
+  setSecret();  
+  selectQuestion();
+  guessCount = 0;
+  guessCounter.innerHTML = guessCount;
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// setting the currentQuestion object when selecting something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  const category = questions.options[questions.selectedIndex].parentNode.label 
+  const value = questions.options[questions.selectedIndex].value 
   currentQuestion = {
-    category: category,
-    // value: value
+    category: category, 
+    value: value, 
   }
+  console.log(category, value)
 }
 
-// This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
+//Function invoked when click on button Find out
+const updateGuesses = () => {
+  guessCount ++; // for each question guesses + 1
+    guessCounter.innerHTML = guessCount; 
+    checkQuestion(); // when select and guess on attribute 
+ };
+
+// Function invoked when click on button Find out
+// Compare currentQuestion/selected attribute details with the secret person
+const checkQuestion = () =>  {
   const { category, value } = currentQuestion
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
+      if (value === secret[category]) {
+        keep = true
+      } else {
+        keep = false
+      }
 
-  } else if (category === 'accessories' || category === 'other') {
-
+  } else if (category === 'accessories' || category === 'other') 
+    { if (secret[category].includes(value)) {
+    keep = true
+  } else {
+    keep = false
   }
```

Comment:
Nice form to make work the "or" option, with the "double if" is a great way to make it work! Good job!